tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -0,0 +1,832 @@
+        {
+            return client
+                .GetDatabase(collectionNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(collectionNamespace.CollectionName);
+        }
+
+        private IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> GetKmsProviders()
+        {
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+
+            var awsRegion = Environment.GetEnvironmentVariable("FLE_AWS_REGION") ?? "us-east-1";
+            var awsAccessKey = Environment.GetEnvironmentVariable("FLE_AWS_ACCESS_KEY_ID") ?? throw new Exception("The AWS_ACCESS_KEY_ID system variable should be configured on the machine.");
+            var awsSecretAccessKey = Environment.GetEnvironmentVariable("FLE_AWS_SECRET_ACCESS_KEY") ?? throw new Exception("The AWS_SECRET_ACCESS_KEY system variable should be configured on the machine.");
+            var kmsOptions = new Dictionary<string, object>
+            {
+                { "region", awsRegion },
+                { "accessKeyId", awsAccessKey },
+                { "secretAccessKey", awsSecretAccessKey }
+            };
+            kmsProviders.Add("aws", kmsOptions);
+
+            var localOptions = new Dictionary<string, object>
+            {
+                { "key", new BsonBinaryData(Convert.FromBase64String(LocalMasterKey)).Bytes }
+            };
+            kmsProviders.Add("local", localOptions);
+
+            return new ReadOnlyDictionary<string, IReadOnlyDictionary<string, object>>(kmsProviders);
+        }
+
+        private Dictionary<string, BsonDocument> GetSchemaMapIfNotNull(BsonDocument schemaMapDocument)
+        {
+            Dictionary<string, BsonDocument> schemaMap = null;
+            if (schemaMapDocument != null)
+            {
+                var element = schemaMapDocument.Single();
+                schemaMap = new Dictionary<string, BsonDocument>
+                    {
+                        { element.Name, element.Value.AsBsonDocument }
+                    };
+            }
+            return schemaMap;
+        }
+
+        private void Insert(
+            IMongoCollection<BsonDocument> collection,
+            bool async,
+            params BsonDocument[] documents)
+        {
+            if (async)
+            {
+                collection
+                    .InsertManyAsync(documents)
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                collection.InsertMany(documents);
+            }
+        }
+
+        public class JsonFileReader : EmbeddedResourceJsonFileReader
+        {
+            #region static
+            private static readonly Lazy<JsonFileReader> __lazyInstance = new Lazy<JsonFileReader>(() => new JsonFileReader(), isThreadSafe: true);
+            public static JsonFileReader Instance => __lazyInstance.Value;
```

Comment:
We don't usually mix fields and properties together.

Reply:
If you are willing to standardize this static #region it would look like this:

    #region static
    // private static fields
    private static readonly string[] __ignoreKeyNames =
    {
        "dbPointer" // not supported
    };
    private static readonly Lazy<JsonFileReader> __instance = new Lazy<JsonFileReader>(() => new JsonFileReader(), isThreadSafe: true);

    // public static properties
    public static JsonFileReader Instance => __instance.Value;
    #endregion

Reply:
done